pkg/platform/runtime/setup/setup.go

```diff
@@ -262,7 +278,12 @@ func (s *Setup) Update(buildResult *apimodel.BuildResult, commit *bpModel.Commit
 	}
 
 	if s.target.ProjectDir() != "" && s.cfg.GetBool(constants.OptinBuildscriptsConfig) {
-		if err := buildscript.Update(s.target, buildResult.AtTime, buildResult.BuildExpression); err != nil {
+		expression, err := buildexpression.New(commit.Expression)
+		if err != nil {
+			return errs.Wrap(err, "failed to parse build expression")
+		}
```

Comment:
Can't you use the existing `expression` defined on line 266?

Reply:
D'oh